scene/gui/foldable_container.cpp

```diff
@@ -0,0 +1,301 @@
+				title_width -= icon->get_width() + h_separation;
+				Point2 icon_pos;
+				if (title_style.is_valid()) {
+					icon_pos = title_style->get_offset();
+				}
+				if (theme_cache.title_font.is_valid()) {
+					if (theme_cache.title_font->get_height(theme_cache.title_font_size) <= icon->get_height()) {
+						title_pos.y += (icon->get_height() - theme_cache.title_font->get_ascent(theme_cache.title_font_size)) / 2;
+					} else {
+						icon_pos.y += (theme_cache.title_font->get_height(theme_cache.title_font_size) - icon->get_height()) / 2;
+					}
+				}
+				icon->draw(ci, icon_pos);
+			}
+
+			if (theme_cache.title_font.is_valid()) {
+				Color font_color = expanded ? theme_cache.title_font_color : theme_cache.title_collapsed_font_color;
+				if (is_hovering) {
+					font_color = theme_cache.title_hover_font_color;
+				}
+				draw_string(theme_cache.title_font, title_pos, title, title_alignment, title_width, theme_cache.title_font_size, font_color);
+			}
+
+			if (expanded && theme_cache.panel_style.is_valid()) {
+				theme_cache.panel_style->draw(ci, Rect2(0, title_size.height, get_size().width, get_size().height - title_size.height));
+			}
+
+			if (has_focus() && theme_cache.focus_style.is_valid()) {
+				theme_cache.focus_style->draw(ci, Rect2(Point2(), get_size()));
+			}
+		} break;
+
+		case NOTIFICATION_SORT_CHILDREN: {
+			Size2 size = get_size();
+			size.height -= title_size.height;
+
+			Point2 ofs;
+			ofs.y += title_size.height;
+
+			if (theme_cache.panel_style.is_valid()) {
+				size -= theme_cache.panel_style->get_minimum_size();
+				ofs += theme_cache.panel_style->get_offset();
+			}
+
+			for (int i = 0; i < get_child_count(); i++) {
+				Control *c = Object::cast_to<Control>(get_child(i));
+				if (!c || !c->is_visible_in_tree()) {
+					continue;
+				}
+				if (c->is_set_as_top_level()) {
+					continue;
+				}
+
+				fit_child_in_rect(c, Rect2(ofs, size));
+			}
+		} break;
+
+		case NOTIFICATION_MOUSE_EXIT: {
+			if (is_hovering) {
+				is_hovering = false;
+				queue_redraw();
+			}
+		} break;
+	}
+}
+
+void FoldableContainer::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_expanded", "expanded"), &FoldableContainer::set_expanded);
+	ClassDB::bind_method(D_METHOD("is_expanded"), &FoldableContainer::is_expanded);
+	ClassDB::bind_method(D_METHOD("set_title", "title"), &FoldableContainer::set_title);
+	ClassDB::bind_method(D_METHOD("get_title"), &FoldableContainer::get_title);
+	ClassDB::bind_method(D_METHOD("set_title_alignment", "alignment"), &FoldableContainer::set_title_alignment);
+	ClassDB::bind_method(D_METHOD("get_title_alignment"), &FoldableContainer::get_title_alignment);
+
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "expanded"), "set_expanded", "is_expanded");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "title"), "set_title", "get_title");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "title_alignment", PROPERTY_HINT_ENUM, "Left,Center,Right"), "set_title_alignment", "get_title_alignment");
```

Comment:
I have ignored fill alignment, since the text already fills the available space automatically, also text clipping happens automatically, but we should follow the `LineEdit` text styling for more advanced text render options.